sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_path_client_async.py

```diff
@@ -0,0 +1,499 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from azure.storage.blob.aio import BlobClient
+from .._shared.base_client_async import AsyncStorageAccountHostsMixin
+from .._path_client import PathClient as PathClientBase
+from .._models import DirectoryProperties
+from .._generated.aio import DataLakeStorageClient
+from ._data_lake_lease_async import DataLakeLeaseClient
+from .._generated.models import StorageErrorException
+from .._deserialize import process_storage_error
+from .._shared.policies_async import ExponentialRetry
+
+
+_ERROR_UNSUPPORTED_METHOD_FOR_ENCRYPTION = (
+    'The require_encryption flag is set, but encryption is not supported'
+    ' for this method.')
+
+
+class PathClient(AsyncStorageAccountHostsMixin, PathClientBase):
+    def __init__(
+            self, account_url,  # type: str
+            file_system_name,  # type: str
+            path_name,  # type: str
+            credential=None,  # type: Optional[Any]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        kwargs['retry_policy'] = kwargs.get('retry_policy') or ExponentialRetry(**kwargs)
+
+        super(PathClient, self).__init__(account_url, file_system_name, path_name, # type: ignore # pylint: disable=specify-parameter-names-in-call
+                                         credential=credential,
+                                         **kwargs)
+
+        kwargs.pop('_hosts', None)
+        self._blob_client = BlobClient(self._blob_account_url, file_system_name, blob_name=path_name,
+                                       credential=credential, _hosts=self._blob_client._hosts, **kwargs)  # type: ignore # pylint: disable=protected-access
+        self._client = DataLakeStorageClient(self.url, file_system_name, path_name, pipeline=self._pipeline)
+        self._loop = kwargs.get('loop', None)
+
+    async def _create(self, resource_type, content_settings=None, metadata=None, **kwargs):
+        # type: (...) -> Dict[str, Union[str, datetime]]
+        """
+        Create directory or file
+
+        :param resource_type: Required for Create File and Create Directory.
```

Comment:
There's some inconsistency in the docstring formatting, should be a new line here (and with `umask`)